functionary/core/utils/minio.py

```diff
@@ -0,0 +1,95 @@
+from datetime import timedelta
+from io import BytesIO
+from os import getenv
+
+from django.http import HttpRequest
+from minio import Minio
+from urllib3.response import HTTPResponse
+
+from core.models import Environment
+
+
+class MinioInterface:
+    def __init__(self, bucket_name: str):
+        self.client = Minio(
+            endpoint=f"{getenv('MINIO_HOST')}:{getenv('MINIO_PORT')}",
```

Comment:
Late follow-up:  The settings should probably reference S3 rather than MINIO.  Also, I'm assuming that region would be required if using actual S3, so that should be an optional config setting here.

Reply:
Fixed